src/components/Table/components/Row.tsx

```diff
@@ -132,245 +135,234 @@ function RowImpl<R extends Kinded, S>(props: RowProps<R>): ReactElement {
   const ref = useRef<HTMLTableRowElement>(null);
 
   return (
-    <RowTag
-      css={rowCss}
-      {...others}
-      data-gridrow
-      {...getCount(row.id)}
+    // we need to use padding for the drag & drop spacing or else drop events don't fire
+    // and we need a containing element for padding to work correctly with the card style row
+    <div
```

Comment:
Wrapping rows in this container seems to be the only sane way to make card style rows work with drag & drop. It doesn't seem to have broken anything wrt rows/tables, at least as far as Beam is concerned. But this is somewhat risky in terms of breaking implementations that rely on Beam. Let me know if this is likely to mess something up.

Reply:
Huh, isn't `RowTag` basically already providing a wrapping div around the cells? It seems odd that we'd need two wrapping divs...

Ah, I get it, both dnd & cards were using padding, but when dragging we want the dnd padding to be "on top" / separate from the card padding?

I'm okay-ish with this as a short-term fix, but yeah let's try and only ad the extra div is dnd is actively being used, just to be extra sure we don't mess existing callers.

Afaiu does dnd use padding to provide the "spacer" effect? Kinda wondering if we could use a psuedo element for that (like the `::before` thing), which I personally haven't done a ton of. Not sure how we'd listen to drop events on that, but maybe we could hook up the drop handlers to the root table div, and that would pick up the `::before` element, maybe encode what got "dropped" using a `data-something` attribute on the psuedo element.

cc @bdow for thoughts.

Reply:
The padding for the spacer goes inside the border of the card, so instead of looking like it's going above or below the row, it looks like it's going inside the card. I will look into the before/after pseudo-elements but I'm pretty sure they won't handle events properly. I could make the container conditional to draggable rows, but theoretically it should be able to work either way.

Reply:
> The padding for the spacer goes inside the border of the card, so instead of
> looking like it's going above or below the row, it looks like it's going inside
> the card.

Yeah, makes sense...

> theoretically it should be able to work either way.

Yeah. Ngl I'm surprised it still works, but I get it, because we're like hand-calcing so much of the column widths/etc; even if it just comes down to DOM aesthetics of avoiding the extra div :shrug: dunno, seems simple enough to avoid/make conditional.

Reply:
conditional it is

Reply:
@stephenh should only affect draggable rows now

Reply:
I was also thinking the pseudo elements would be the solution here instead of a wrapping element. I'm surprised they don't.

Reply:
@bsholmes nice, thank you!